src/compose/resolve-flow-collection.ts

```diff
@@ -18,13 +19,17 @@ export function resolveFlowCollection(
   { composeNode, composeEmptyNode }: ComposeNode,
   ctx: ComposeContext,
   fc: FlowCollection,
-  onError: ComposeErrorHandler
+  onError: ComposeErrorHandler,
+  tag?: CollectionTag
 ) {
   const isMap = fc.start.source === '{'
   const fcName = isMap ? 'flow map' : 'flow sequence'
+  let NodeClass = tag?.nodeClass
+  const MapClass = NodeClass || YAMLMap
+  const SeqClass = NodeClass || YAMLSeq
   const coll = isMap
-    ? (new YAMLMap(ctx.schema) as YAMLMap.Parsed)
-    : (new YAMLSeq(ctx.schema) as YAMLSeq.Parsed)
+    ? (new MapClass(ctx.schema) as YAMLMap.Parsed)
+    : (new SeqClass(ctx.schema) as YAMLSeq.Parsed)
```

Comment:
That was to avoid a typescript complaint:

<img width="742" alt="Screenshot 2023-04-29 at 13 29 22" src="https://user-images.githubusercontent.com/9287/235323007-30a17c2a-c14f-4c51-aefb-0df8e257136a.png">

If you think this alternative is less convoluted, it would also work:

```ts
  let NodeClass = tag?.nodeClass ?? (isMap ? YAMLMap : YAMLSeq)
  const coll:YAMLMap.Parsed | YAMLSeq.Parsed = new (NodeClass as {
    new(ctx: ComposeContext['schema']): YAMLMap.Parsed | YAMLSeq.Parsed
  })(ctx.schema)
```

Reply:
or this:

```ts
  let NodeClass = (tag?.nodeClass ?? (isMap ? YAMLMap : YAMLSeq)) as {
    new (ctx: ComposeContext['schema']): YAMLMap.Parsed | YAMLSeq.Parsed
  }
  const coll: YAMLMap.Parsed | YAMLSeq.Parsed = new NodeClass(ctx.schema)
```

Reply:
Either this is just TS being difficiult, in which case preferably the latter, unless this works?
```ts
let NodeClass = (tag?.nodeClass ?? (isMap ? YAMLMap : YAMLSeq)) as (
  typeof YAMLMap.Parsed | typeof YAMLSeq.Parsed
)
const coll = new NodeClass(ctx.schema)
```

But if it's not, see the suggestion in `src/schema/types.ts` -- does adding that resolve this TS error without adding a typecast?

Reply:
You do need a typecast, because the `NodeClass` might (according to TS) be a type that doesn't have the `flow`, `comment` etc fields.

Casting as `typeof YAMLMap.Parsed` doesn't work, because in that context, `YAMLMap` is a namespace, and `Parsed` is an interface.  So `typeof YAMLMap` points to the _class_ definition, which doesn't have a `Parsed` member.

But this works, and imo fairly clear:

```ts
  let NodeClass = (tag?.nodeClass ?? (isMap ? YAMLMap : YAMLSeq)) as {
    new (schema: Schema): YAMLMap.Parsed | YAMLSeq.Parsed
  }
```

Reply:
To avoid the typecast, I think we'd have to more rigidly define the interface that the NodeClass needs to provide, which could be annoying in practice. Probably better to just document that it should extend YAMLMap or YAMLSeq, and if someone goes off the paved path and implements it their own way, oh well.